rpcserver.go

```diff
@@ -2483,6 +2483,98 @@ func (s *rpcServer) DecodeSidecarTicket(ctx context.Context,
 	return marshallTicket(ticket), nil
 }
 
+// ListSidecars lists all sidecar tickets currently in the local database. This
+// includes tickets offered by our node as well as tickets that our node is the
+// recipient of.
+func (s *rpcServer) ListSidecars(_ context.Context,
+	_ *poolrpc.ListSidecarsRequest) (*poolrpc.ListSidecarsResponse,
+	error) {
+
+	tickets, err := s.server.db.Sidecars()
+	if err != nil {
+		return nil, fmt.Errorf("error reading sidecar tickets: %v", err)
+	}
+
+	resp := &poolrpc.ListSidecarsResponse{
+		Tickets: make([]*poolrpc.DecodedSidecarTicket, len(tickets)),
+	}
+	for idx, ticket := range tickets {
+		resp.Tickets[idx] = marshallTicket(ticket)
+	}
+
+	return resp, nil
+}
+
+// CancelSidecar cancels the execution of a specific sidecar ticket. Depending
+// on the state of the sidecar ticket its associated bid order might be
+// canceled as well (if this ticket was offered by our node).
+func (s *rpcServer) CancelSidecar(ctx context.Context,
+	req *poolrpc.CancelSidecarRequest) (*poolrpc.CancelSidecarResponse,
+	error) {
+
+	tickets, err := s.server.db.Sidecars()
```

Comment:
I think it'll be worth it to just expose a proper method here to index into the DB vs doing this linear scan. Some of the upcoming "pool as a service" systems may end up having quite a large amount of tickets in the database here. 

I think we can get away w/ a cursor seek, since the key has a prefix of the sidecar ID within the bucket itself.

Reply:
Yeah, makes sense. I added the seek by ID method. This should speed things up. And this also allows us to filter by ID in the `ListSidecars` RPC.